boards/st/nucleo_g431kb/doc/index.rst

```diff
@@ -0,0 +1,189 @@
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/st/nucleo_g431kb/nucleo_g431kb_defconfig`
+
+
+Connections and IOs
+===================
+
+Nucleo G431KB Board has 6 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For more details please refer to `STM32G4 Nucleo-32 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+.. rst-class:: rst-columns
+
+- LPUART_1_TX : PA2
+- LPUART_1_RX : PA3
+- LD2 : PB8
+- PWM_4_CH_3 : PB8
+- I2C_2_SCL : PA9
+- I2C_2_SDA : PA8
+
+System Clock
+------------
+
+The Nucleo G431KB System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default the external oscillator is not connected to the board. Therefore only the internal
+High Speed oscillator is supported. By default System clock is driven by PLL clock at 170MHz,
+the PLL is driven by the 16MHz high speed internal oscillator.
+
+Serial Port
+-----------
+
+Nucleo G431KB board has 1 U(S)ARTs and one LPUART. The Zephyr console output is assigned to LPUART1.
+Default settings are 115200 8N1.
+
+Please note that LPUART1 baudrate is limited to 9600 if the MCU is clocked by LSE (32.768 kHz) in
+low power mode.
+
+Programming and Debugging
+*************************
+
+Nucleo G431KB Board includes an ST-Link/V3 embedded debug tool interface.
+
+Applications for the ``nucleo_g431kb`` board target can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+The board is configured to be flashed using west `STM32CubeProgrammer`_ runner,
+so its :ref:`installation <stm32cubeprog-flash-host-tools>` is required.
+
+Alternatively, OpenOCD, or pyOCD can also be used to flash the board using
+the ``--runner`` (or ``-r``) option:
+
+.. code-block:: console
+
+   $ west flash --runner openocd
+   $ west flash --runner pyocd
+
+To enable support of the stm32g342kb SoC in pyocd the pyocd pack for this soc has to be installed first:
```

Comment:
```suggestion
To enable support for STM32G431KB SoC in pyOCD, its pack has to be installed first:
```
Have you verified that this pack is actually available?

Reply:
Yes, i use the configuration since you suggested to use the pyocd runner as default. I just ckecked it again, it gets pulled in via Keil.STM32G4xx_DFP.1.6.0